addons/mail/static/src/core/common/composer.js

```diff
@@ -735,7 +735,10 @@ export class Composer extends Component {
         } else {
             this.env.services.dialog.add(MessageConfirmDialog, {
                 message: composer.message,
-                onConfirm: () => this.message.remove(),
+                onConfirm: () => {
+                    this.message.remove();
+                    this.props.onDiscardCallback();
```

Comment:
`onDiscardCallback` is documented as optional, so should be defensive with optional chaining `onDiscardCallback?.()`